site/app/libraries/database/DatabaseQueries.php

```diff
@@ -813,9 +813,25 @@ public function removeNotificationsPost($post_id) {
         $this->course_db->query("DELETE FROM notifications where metadata::json->>'post_id' = ?", [$post_id]);
     }
 
-    public function isStaffPost($author_id) {
+    public function isStaffPost(string $author_id): bool {
         $this->course_db->query("SELECT user_group FROM users WHERE user_id=?", [$author_id]);
-        return intval($this->course_db->rows()[0]['user_group']) <= 3;
+        return intval($this->course_db->row()['user_group']) <= User::GROUP_LIMITED_ACCESS_GRADER;
+    }
+
+    public function isLimitedAccessGraderPost(string $author_id): bool {
+        $this->course_db->query("SELECT user_group FROM users WHERE user_id=?", [$author_id]);
+        return intval($this->course_db->row()['user_group']) === User::GROUP_LIMITED_ACCESS_GRADER;
+    }
+
+    public function isInstructorOrFullAccessGraderPost(string $author_id): bool {
+        $this->course_db->query("SELECT user_group FROM users WHERE user_id=?", [$author_id]);
+        return intval($this->course_db->row()['user_group']) <= User::GROUP_FULL_ACCESS_GRADER;
+    }
+
+    public function getAuthorUserGroups(array $author_ids): array {
+        $placeholders = $this->createParamaterList(count($author_ids));
+        $this->course_db->query("SELECT user_id, user_group FROM users WHERE user_id IN {$placeholders}", $author_ids);
+        return $this->course_db->rows();
     }
```

Comment:
I believe that all of these functions are unnecessary.  It would be better to use the relevant functions in `site/app/models/User.php`.